tfjs-backend-webgpu/src/shader_util.ts

```diff
@@ -32,3 +32,24 @@ export function symbolicallyComputeStrides(
 
   return strides;
 }
+
+// atomicAdd only supports uint/int type. For float, we use
+// atomicCompareExchangeWeak to simulate.
+export const atomicAddSnippet =
+    (dst: string, index: string, val: string, component: number) => {
+      return `
+    for (var i = 0; i < ${component}; i = i + 1) {
+      var oldValue = atomicLoad(&(${dst}[${index} + i]));
+      var exchanged = false;
+      for (; !exchanged;) {
```

Comment:
I think the initialization of `exchanged` is a waste of cycles. Can we rewrite the loop to

```rust
loop {
  if exchanged {
    break;
  }
}
```